terragrunt/modules/crates-io/compute-static/src/main.rs

```diff
@@ -127,6 +130,28 @@ fn rewrite_urls_with_plus_character(request: &mut Request) {
     }
 }
 
+/// Rewrite `/crates/{crate}/{version}/download` URLs to
+/// `/crates/{crate}/{crate}-{version}.crate`
+///
+/// cargo versions before 1.24 don't support placeholders in the `dl` field
+/// of the index, so we need to rewrite the download URL to point to the
+/// crate file instead.
+fn rewrite_download_urls(request: &mut Request) {
+    static RE: Lazy<Regex> = Lazy::new(|| {
+        Regex::new(r"^/crates/(?P<crate>[^/]+)/(?P<version>[^/]+)/download$").unwrap()
+    });
+
+    let url = request.get_url_mut();
+    let path = url.path();
+
+    if let Some(captures) = RE.captures(path) {
+        let krate = captures.name("crate").unwrap().as_str();
```

Comment:
Unwrapping here seems dangerous. I'm not sure how much debugging information we'll get if we just fail hard. Instead, I think it is better to change the return type of the function to `Result<(), Error>` and handle the error gracefully. And if we log an error with `error!`, we'll get to see that in the service logs.

Reply:
hmm, can you think of a scenario where the unwrap would panic? the outer if let ensures that the regex matches and both of the captures are non-optional so I think unwrap is safe here for this regex 🤔

Reply:
That's a good argument. I don't think this can panic in its current form, but I'm also somewhat paranoid about changes in the future and other people assuming, based on this example, that it's okay to `unwrap` in this binary.